app/src/lib/stores/app-store.ts

```diff
@@ -6215,21 +6150,36 @@ export class AppStore extends TypedBaseStore<IAppState> {
       )
     }
 
-    this.repositoryStateCache.updateCherryPickState(repository, () => {
-      return {
-        progress: {
-          kind: 'multiCommitOperation',
-          value: 0,
-          position: 1,
-          totalCommitCount: commits.length,
-          currentCommitSummary: commits[commits.length - 1].summary,
-        },
+    this.repositoryStateCache.updateMultiCommitOperationState(
+      repository,
+      () => {
+        return {
+          progress: {
+            kind: 'multiCommitOperation',
+            value: 0,
+            position: 1,
+            totalCommitCount: commits.length,
+            currentCommitSummary: commits[commits.length - 1].summary,
+          },
+        }
       }
-    })
+    )
 
     this.emitUpdate()
   }
 
+  private getMCOProgressCallBack(repository: Repository) {
```

Comment:
Not a fan of `MCO`, but I understand why 😂 

However keeping `MCO` could make reading the code a bit harder for someone who doesn't know about this? (poor Markus 🤣 )

Reply:
Yeah... I know.. I will switch it back. :D